consensus/quorum/quorum.go

```diff
@@ -246,6 +250,41 @@ func (s *cIdentities) NthNextHmy(instance shardingconfig.Instance, pubKey *bls.P
 	return found, &s.publicKeys[idx]
 }
 
+// NthNextHmy return the Nth next pubkey of Harmony + allowlist nodes, next can be negative number
+func (s *cIdentities) NthNextHmyExt(instance shardingconfig.Instance, pubKey *bls.PublicKeyWrapper, next int) (bool, *bls.PublicKeyWrapper) {
+	found := false
+
+	idx := s.IndexOf(pubKey.Bytes)
+	if idx != -1 {
+		found = true
+	}
+	numHmyNodes := instance.NumHarmonyOperatedNodesPerShard()
+	// sanity check to avoid out of bound access
+	if numHmyNodes <= 0 || numHmyNodes > len(s.publicKeys) {
+		numHmyNodes = len(s.publicKeys)
+	}
+	nth := idx
+	if idx >= numHmyNodes {
+		nth = sort.SearchInts(s.allowlistIndex, idx) + numHmyNodes
+	}
+
+	numExtNodes := instance.ExternalAllowlistLimit()
+	if numExtNodes > len(s.allowlistIndex) {
+		numExtNodes = len(s.allowlistIndex)
+	}
+
+	totalNodes := numHmyNodes + numExtNodes
+	// (totalNodes + next%totalNodes) can convert negitive 'next' to positive
```

Comment:
How can next be negative?

Reply:
The comment for the function 'NthNextHmy()' says that `next` can be negative number.